tools/cldr-code/src/main/java/org/unicode/cldr/test/CheckDates.java

```diff
@@ -641,6 +643,99 @@ && isTooMuchWiderThan(value, abbrValue)
         return this;
     }
 
+    static final Set<Integer> expectedField =
+            Set.of(
+                    DateTimePatternGenerator.YEAR,
+                    DateTimePatternGenerator.QUARTER,
+                    DateTimePatternGenerator.MONTH,
+                    DateTimePatternGenerator.DAY,
+                    DateTimePatternGenerator.WEEKDAY,
+                    DateTimePatternGenerator.HOUR,
+                    DateTimePatternGenerator.MINUTE,
+                    DateTimePatternGenerator.SECOND,
+                    DateTimePatternGenerator.ZONE);
+
+    /**
+     * Returns null if the path is not a calendar path for iso8601, or if it is ok for iso8601.<br>
+     * Otherwise returns a string with the error.
+     *
+     * @param path
+     * @param value
+     * @return
+     */
+    // This is public for testing
+
+    public static String checkIso8601(String path, String value) {
+        // ldml/dates/calendars/calendars/dateFormats/dateFormatLength
+        XPathParts parts = XPathParts.getFrozenInstance(path);
+        if (!"iso8601".equals(parts.getAttributeValue(3, "type"))) {
+            return null;
+        }
+        String key = parts.getElement(5);
+        boolean isInterval = false;
+        switch (key) {
+            case "dateTimeFormatLength":
+            case "appendItem":
+                int index0 = value.indexOf("{0}");
+                int index1 = value.indexOf("{1}");
+                if (index0 > index1) {
+                    return "Field out of order: {1}…{0}";
+                }
+                return null;
+            case "dateFormatLength":
+            case "timeFormatLength":
+            case "availableFormats":
+                break;
+            case "intervalFormats":
+                isInterval = true;
+                break;
+            default:
+                return null;
+        }
+        // verify the order is year month dow hour minute second
```

Comment:
dow = day?

Reply:
dow = day of week

It's restricting it to `expectedFields`.    So in order, YEAR QUARTER MONTH WEEKDAY(dow) DAY HOUR MINUTE SECOND ZONE

```suggestion
        // verify the order is YEAR QUARTER MONTH WEEKDAY(dow) DAY HOUR MINUTE SECOND ZONE (by the integer values)
```

Reply:
Day Of Week (will fix in later PR)